test/index.js

```diff
@@ -23,22 +35,116 @@ describe('superagent-valence', function() {
 	});
 
 	it('adds csrf token for relative URLs', function() {
+		global.D2LAccessTokenExpiresAt = theFuture();
+
 		var endpoint = nock('http://localhost')
-			.matchHeader('X-D2L-App-Id', /.*/)
-			.matchHeader('X-Csrf-Token', CSRF_TOKEN)
-			.get('/url')
+			.matchHeader('X-Csrf-Token', XSRF_TOKEN)
+			.get('/api')
 			.reply(200);
 
-		request.get('/url')
+		request
+			.get('/api')
 			.use(valence)
 			.end(function() {});
 
 		endpoint.done();
 	});
 
-	it('does not add csrf token for non-relative URLs', function() {
-		var req = request.get('http://localhost/url').use(valence);
+	it('does not add xsrf token for non-relative URLs', function() {
+		var req = request.get('http://localhost/api').use(valence);
 
 		should.not.exist(req.header['X-Csrf-Token']);
+		req.end.should.equal(Object.getPrototypeOf(req).end); // no funny business
+	});
+
+	it('sends refreshcookie preflight on boot', function() {
+		global.D2LAccessTokenExpiresAt = 0;
+
+		var endpoint = nock('http://localhost')
+			.post('/d2l/lp/auth/oauth2/refreshcookie')
+			.matchHeader('X-Csrf-Token', XSRF_TOKEN)
+			.reply(204)
+			.get('/api')
+			.matchHeader('X-Csrf-Token', XSRF_TOKEN)
+			.reply(200);
+
+		request
+			.get('/api')
+			.use(valence)
+			.end(function() {
+				endpoint.done();
+
+				global.D2LAccessTokenExpiresAt
+					.should.equal(0); // no cache-control --> can't set an expiry
```

Comment:
Huh. I do not know. I fiddled with the assertions and it seemed to be working. I've added the `done()`s though because I'm guessing it was just "luck"?